app/src/main/java/com/example/sharewhatyoucanproject/rdashboard/rdetailedPosts/RdetailsViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.sharewhatyoucanproject.rdashboard.rdetailedPosts
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.sharewhatyoucanproject.models.PostModel
+import com.google.firebase.firestore.FirebaseFirestore
+
+class RDetailsViewModel(
+    private val db: FirebaseFirestore,
+) : ViewModel()
+
+class RDetailsViewModelFactory(
+    private val db: FirebaseFirestore = FirebaseFirestore.getInstance(),
+) : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
+        return RDetailsViewModel(db) as T
+    }
+}
+
+sealed class DetailsResult {
+    data class Success(val arrayList: ArrayList<PostModel>) : DetailsResult()
```

Comment:
why does the success object for the details view need a list of posts? just making sure I understand

Reply:
Sorry, I will be creating different model for receiver post. I took reference of PostlistViewModel, that's why it's little similar to that viewmodel. I will change it.